test/conftest.py

```diff
@@ -734,3 +777,144 @@ def sampleset_ref(clients: dict[str, Any], ws_id: int, reads_refs: list[str]) ->
+    """Function to return a function that creates fixtures."""
+
+    @pytest.fixture(scope="session", name=set_name)
+    def set_fixture(
+        request: FixtureRequest,
+        set_api_client: SetAPI,
+        context: dict[str, Any],
+        ws_id: int,
+    ) -> dict[str, Any]:
+        """Create named fixtures; the name is denoted by the `set_name`.
+
+        :param request: fixture parameters
+        :type request: FixtureRequest
+        :param set_api_client: SetAPI client
+        :type set_api_client: SetAPI
+        :param context: KBase context
+        :type context: dict[str, Any]
+        :param ws_id: workspace ID
+        :type ws_id: int
+        :return: output of save_set / save_empty_set
+        :rtype: dict[str, Any]
+        """
+        refs = None
+        # if "refs" is populated, use request.getfixturevalue to retrieve the fixture by name
+        if fixture_info.get("refs"):
+            refs = request.getfixturevalue(fixture_info["refs"])
+
+        return set_factory(
+            fixture_info["set_item_name"],
+            refs,
+            set_api_client,
+            context,
+            ws_id,
+            fixture_info.get("name_qualifier"),
+        )
+
+    return set_fixture
+
+
+# mapping of fixture names to the details of the fixture; these include
+# - set_item_name of the contents (for inserting into 'save_***_set_v1' when saving the set)
+# - refs (where appropriate; the items to go in the set)
+# - name_qualifier (to differentiate between sets of the same type with different contents)
+
+SET_FIXTURE_MAP = {
+    "assembly_set": {"set_item_name": ASSEMBLY, "refs": "assembly_refs"},
+    "empty_assembly_set": {"set_item_name": ASSEMBLY},
+    "differential_expression_matrix_no_genome_set": {
+        "set_item_name": DIFFERENTIAL_EXPRESSION_MATRIX,
+        "refs": "differential_expression_matrix_no_genome_refs",
+        "name_qualifier": "no_genome",
+    },
+    "differential_expression_matrix_with_genome_set": {
+        "set_item_name": DIFFERENTIAL_EXPRESSION_MATRIX,
+        "refs": "differential_expression_matrix_with_genome_refs",
+        "name_qualifier": "with_genome",
+    },
+    "expression_set": {"set_item_name": EXPRESSION, "refs": "expression_refs"},
+    "feature_set_set": {"set_item_name": FEATURE_SET, "refs": "feature_set_refs"},
+    "empty_feature_set_set": {"set_item_name": FEATURE_SET},
+    "genome_set": {"set_item_name": GENOME, "refs": "genome_refs"},
+    "empty_genome_set": {"set_item_name": GENOME},
+    "reads_set": {"set_item_name": READS, "refs": "reads_refs"},
+    "empty_reads_set": {"set_item_name": READS},
+    "reads_alignment_set": {
+        "set_item_name": READS_ALIGNMENT,
+        "refs": "alignment_refs",
+    },
+}
+
+# Create a fixture for each set type in the fixture map
+for set_name, fixture_info in SET_FIXTURE_MAP.items():
+    globals()[set_name] = create_set_fixture(set_name, fixture_info)
```

Comment:
this is the ugly bit that actually saves the fixtures named in SET_FIXTURE_MAP into the global namespace so that pytest can use them throughout the tests